libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx

```diff
@@ -171,11 +182,47 @@ export interface IDashboardProps {
 }
 
 const DashboardInner: React.FC<IDashboardProps> = (props: IDashboardProps) => {
-    const { dashboardRef, dashboardLayoutConfig, drillableItems, ErrorComponent, LoadingComponent } = props;
+    const {
+        dashboardRef,
+        dashboardLayoutConfig,
+        drillableItems,
+        filterBarConfig,
+        ErrorComponent,
+        LoadingComponent,
+    } = props;
     const customLayout = typeof props.children === "function" ? props.children?.(null) : props.children;
     const LayoutComponent = dashboardLayoutConfig?.Component ?? Layout;
+    const FilterBarComponent = filterBarConfig?.Component ?? FilterBar;
+
+    const filters = useDashboardSelector(selectFilterContextFilters);
+    const dispatch = useDashboardDispatch();
+
     return (
         <React.Fragment>
+            <FilterBarComponent
+                filters={filters}
+                onFilterChanged={(filter) => {
+                    if (!filter) {
+                        // all time filter
+                        dispatch(clearDateFilterSelection());
+                    } else if (isDashboardDateFilter(filter)) {
+                        const { type, granularity, from, to } = filter.dateFilter;
+                        dispatch(changeDateFilterSelection(type, granularity, from, to));
+                    } else if (isDashboardAttributeFilter(filter)) {
+                        const { attributeElements, negativeSelection, localIdentifier } =
+                            filter.attributeFilter;
+                        dispatch(
+                            changeAttributeFilterSelection(
+                                localIdentifier!,
+                                attributeElements,
+                                negativeSelection ? "NOT_IN" : "IN",
+                            ),
+                        );
+                    } else {
+                        invariant(false, "Unknown filter type");
+                    }
```

Comment:
i propose to move this into a separate function.. keep the dashboard component small

Reply:
Moved to a custom hook.